neuromation/cli/click_types.py

```diff
@@ -193,3 +194,41 @@ async def async_complete(
 
 
 PRESET = PresetType()
+
+
+class JobType(AsyncType[str]):
+    name = "job"
+
+    async def async_convert(
+        self,
+        root: Root,
+        value: str,
+        param: Optional[click.Parameter],
+        ctx: Optional[click.Context],
+    ) -> str:
+        return value
+
+    async def async_complete(
+        self, root: Root, ctx: click.Context, args: Sequence[str], incomplete: str
+    ) -> List[Tuple[str, Optional[str]]]:
+        async with await root.init_client() as client:
+            ret: List[Tuple[str, Optional[str]]] = []
+            now = datetime.now()
+            async for job in client.jobs.list(
+                since=now - timedelta(days=7), reverse=True, limit=100
+            ):
+                job_name = job.name or ""
+                for test in (
+                    job.id,
+                    job_name,
+                    f"job:{job.id}",
+                    f"job:/{client.username}/{job.id}",
+                    f"job://{client.cluster_name}/{client.username}/{job.id}",
+                ):
+                    if test.startswith(incomplete):
+                        ret.append((test, job_name))
```

Comment:
What is the meaning of this tuple? I have not find the documentation.

Reply:
The first item is the found match, the second is for help string (used in `zsh` completion only).

Reply:
I've found that just job ID doesn't help too much to select the job.
The help text can contain more info, suggestions are welcome!

Reply:
Then you should return `job_name` as the first item if `job_name.startswith(incomplete)` and an URI if `incomplete` matches `job:...`.